clouddriver-cf/src/main/groovy/com/netflix/spinnaker/clouddriver/cf/provider/config/CloudFoundryProviderConfig.groovy

```diff
@@ -0,0 +1,108 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cf.provider.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.cats.agent.Agent
+import com.netflix.spinnaker.cats.agent.CachingAgent
+import com.netflix.spinnaker.cats.provider.ProviderSynchronizerTypeWrapper
+import com.netflix.spinnaker.clouddriver.cf.CloudFoundryCloudProvider
+import com.netflix.spinnaker.clouddriver.cf.provider.CloudFoundryProvider
+import com.netflix.spinnaker.clouddriver.cf.provider.agent.ClusterCachingAgent
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.utils.CloudFoundryClientFactory
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.DependsOn
+import org.springframework.context.annotation.Scope
+
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * @author Greg Turnquist
+ */
+@Configuration
+class CloudFoundryProviderConfig {
+
+  @Bean
+  @DependsOn('cloudFoundryAccountCredentials')
+  CloudFoundryProvider cloudFoundryProvider(CloudFoundryCloudProvider cloudFoundryCloudProvider,
+                                            AccountCredentialsRepository accountCredentialsRepository,
+                                            CloudFoundryClientFactory cloudFoundryClientFactory,
+                                            ObjectMapper objectMapper,
+                                            ApplicationContext ctx,
+                                            Registry registry) {
+    def cloudFoundryProvider =
+        new CloudFoundryProvider(cloudFoundryCloudProvider, Collections.newSetFromMap(new ConcurrentHashMap<Agent, Boolean>()))
+
+    synchronizeCloudFoundryProvider(
+        cloudFoundryProvider,
+        cloudFoundryCloudProvider,
+        accountCredentialsRepository,
+        cloudFoundryClientFactory,
+        objectMapper,
+        ctx,
+        registry)
+
+    cloudFoundryProvider
+  }
+
+  @Bean
+  CloudFoundrySynchronizerTypeWrapper cloudFoundrySynchronizerTypeWrapper() {
+    new CloudFoundrySynchronizerTypeWrapper()
+  }
+
+  class CloudFoundrySynchronizerTypeWrapper implements ProviderSynchronizerTypeWrapper {
+    @Override
+    Class getSynchronizerType() {
+      CloudFoundryProviderSynchronizer
+    }
+  }
+
+  class CloudFoundryProviderSynchronizer {}
+
+  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
```

Comment:
> P.S. This is a feature that works out of the box with Spring CLoud Config Server.

Not really - the interesting bit here is unscheduling and rescheduling the caching agent processes. This code will be nearly identical if we bring in spring cloud config with the exception of changing the current prototype_scopes to refresh_scope